askama_shared/src/generator.rs

```diff
@@ -1722,7 +1722,10 @@ impl MapChain<'_, &str, LocalMeta> {
 
     fn resolve_or_self(&self, name: &str) -> String {
         self.resolve(name)
-            .unwrap_or_else(|| format!("self.{}", name))
```

Comment:
I think this would be easier to follow as a three-arm match expression.

Reply:
You mean like this?

```rust
match self.resolve(name) {
    Some(name) => name,
    None if name.chars().any(char::is_uppercase) => name.to_string(),
    None => format!("self.{}", name),
}
```

Reply:
Yup!